src/exercise/DiscountExercise.sol

```diff
@@ -182,6 +273,10 @@ contract DiscountExercise is BaseExercise {
     /// @notice Returns the amount of payment tokens required to exercise the given amount of options tokens.
     /// @param amount The amount of options tokens to exercise
     function getPaymentAmount(uint256 amount) external view returns (uint256 paymentAmount) {
-        paymentAmount = amount.mulWadUp(oracle.getPrice().mulDivUp(multiplier, MULTIPLIER_DENOM));
+        return _getPaymentAmount(amount);
+    }
+
+    function _getPaymentAmount(uint256 amount) private view returns (uint256 paymentAmount) {
```

Comment:
think about if internal or not once more (namespace vs possibility of wanting to call this function in child)